src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -221,6 +221,35 @@ private void checkIfGroupsExists(Course course) {
         return courses;
     }
 
+    /**
+     * GET /courses/:id/for-dashboard
+     *
+     * @param id the id of the course to retrieve
+     * @return data about a course including all exercises, plus some data for the tutor
+     * as tutor status for assessment
+     */
+    @GetMapping("/courses/{id}/for-tutor-dashboard")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Course> getCourseForTutorDashboard(Principal principal, @PathVariable Long id) {
+        log.debug("REST request /courses/{id}/for-tutor-dashboard");
+        Course course = courseService.findOne(id);
+        if (!userHasPermission(course)) return forbidden();
+
+        User user = userService.getUserWithGroupsAndAuthorities();
+        List<Exercise> exercises = exerciseService.findAllForCourse(course, false, principal, user);
+
+        for (Exercise exercise: exercises) {
```

Comment:
This can get expensive for many exercises. Can you have another look at the `getAllCoursesForDashboard(...)` method? Here, they do one database call for all participations. Hence it would be great, if you can rework the below code, so that all tutorParticipations are loaded in one DB call

Reply:
I will completely rework this code in the next days, we need a lot less information in the course dashboard actually

Reply:
I've refactored all the method, now I do only one call to retrieve all the participations